### PR TITLE
Add php8.1 build

### PR DESCRIPTION
**Pull request by icanhazstring**
Add PHP8.1 Build for CI

### CODE REVIEW

**Comment by icanhazstring** [comment]
Currently blocked by:
https://github.com/php-ds/ext-ds/issues/188

File: composer.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
         }
     ],
     "require": {
-        "php": ">=7.3 || ^8.0",
+        "php": ">=7.3",
```
Comment (simPod) [inline composer.json:20]: Do we want to open it so there's not upper bound and it installs on PHP 15?

**Comment by simPod** [comment]
> Currently blocked by: php-ds/ext-ds#188 as the local docker setup does not work.

We can compile it from master if it's needed

File: composer.json (hunk 1 above)
Comment (icanhazstring) [reply composer.json:20]: Good point 😉

**Comment by icanhazstring** [comment]
> We can compile it from master if it's needed

Actually no need. The build itself works, so the lib is working for 8.1. It is just a problem for local dev using docker.

File: composer.json (hunk 1 above)
Comment (simPod) [reply composer.json:20]: But I think it's already broken by having `>=7.3` so there's not much we can do about the underlying issue.

**Pull request merged**